### PR TITLE
Bump to rails 6 and ruby 2.7

### PR DESCRIPTION
**Pull request by leequarella**
Rails is deprecating the use of `content_type` and replacing it with `content_mime_type` https://github.com/rails/rails/commit/840551307139324300b943901d84cbacba11862d

This upgrade is needed in order to keep the monolith up to date with Rails `main`.

### CODE REVIEW

**Comment by leequarella** [comment]
As of https://github.com/github/github/pull/168859 dotcom is running on this branch in production.

**Comment by bbasata** [comment]
Tagging @rnkaufman for review.

At a glance, I imagine it would be helpful to update the gemspec to require Rails 6.

File: spec/dummy/config/environments/test.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@
   config.eager_load = false
 
   # Configure static file server for tests with Cache-Control for performance.
-  if Rails.version.starts_with?("5")
+  if Rails.version.start_with?("5")
```
Comment (rnkaufman) [inline spec/dummy/config/environments/test.rb:16]: I think we might still want `Cache-Control` to be set for Rails 6 as well now, do we?

Also, I searched through the repo and I found several other deprecated `starts_with?` calls within the specs. Do you mind checking for those and updating them as well?

**Review by rnkaufman — APPROVED** [review]
I left a small comment regarding some of the spec files. I also updated the gemspec as per @bbasata's observation.

I performed some testing today with some dependent apps that I help maintain, the functionality appears to be the same. 👍🏻  I feel comfortable merging these changes when the comment is addressed.

**Comment by dbussink** [comment]
I reduced dependencies in #55 but it might be good to add that to this PR instead.

**Review by dbussink — APPROVED** [review]

**Pull request merged**